packages/flutter_tools/test/commands.shard/hermetic/attach_test.dart

```diff
@@ -144,114 +137,6 @@ void main() {
         Logger: () => logger,
       });
 
-      testUsingContext('finds all observatory ports and forwards them', () async {
```

Comment:
This test causes a Dart native crash on my computer. I also had tremendous difficulty updating it due to the level of mocking.

I think this needs to be re-imagined as an integration test

Reply:
This is covered by https://github.com/flutter/flutter/issues/46180 , it just turns out more than one test is crashing :( 

I think fixing this test requires some more globals removal and refactoring of the resident runner. Then, the test should really just be a unit test of the stream reconnect logic.

There are some other issues that need to be covered as well:

https://github.com/flutter/flutter/issues/55864